03_letters/luliangLi/main_test.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"bytes"
+	"reflect"
+	"strconv"
+	"testing"
+)
+
+func TestSortLetters(t *testing.T) {
+
+	testCases := map[string]struct {
+		input string
+		want  []string
+	}{
+		"normal": {input: "aba", want: []string{"a:2", "b:1"}},
+		"blank":  {input: "", want: []string{}},
+		"long":   {input: "aabbcccdddd", want: []string{"d:4", "c:3", "a:2", "b:2"}},
+		"single": {input: "aaaa", want: []string{"a:4"}},
+	}
+
+	for name, test := range testCases {
+		// t.Run creates a sub test and runs it like a normal test
+		test := test
+
+		t.Run(name, func(t *testing.T) {
+			actual := sortLetters(letters(test.input))
```

Comment:
I think it is best to test your functions in isolation, that way if you get a failure, you can be sure what function is causing the underlying issue. Testing the concatenation of the two functions is sort of achieved in the main anyway, which acts as your integration test.

Please break this down into 2 tests.
```go
func TestSortLetters(t *testing.T) {
    ...
}

func TestLetters(t *testing.T) {
    ...
}
```